test/src/test/java/aima/test/unit/search/uninformed/BidirectionalSearchTest.java

```diff
@@ -29,50 +40,57 @@
-        Assert.assertEquals(
-                Arrays.asList(new GoAction(SimplifiedRoadMapOfPartOfRomania.BUCHAREST)),
-                searchForActions(problemPairtest3.getFirst(), problemPairtest3.getSecond()).fromGoalStatePart());
-
-        Pair<Problem<GoAction, InState>, Problem<GoAction, InState>> problemPairtest4 = ProblemFactory.getSimpleBidirectionalSearchProblem(
-                SimplifiedRoadMapOfPartOfRomania.TIMISOARA, SimplifiedRoadMapOfPartOfRomania.URZICENI);
-        Assert.assertEquals(
-                Arrays.asList(new GoAction(SimplifiedRoadMapOfPartOfRomania.ARAD),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.SIBIU),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.FAGARAS),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.BUCHAREST)),
-                searchForActions(problemPairtest4.getFirst(), problemPairtest4.getSecond()).fromInitialStateToGoalState());
-
-        Pair<Problem<GoAction, InState>, Problem<GoAction, InState>> problemPairtest5 = ProblemFactory.getSimpleBidirectionalSearchProblem(
-                SimplifiedRoadMapOfPartOfRomania.TIMISOARA, SimplifiedRoadMapOfPartOfRomania.URZICENI);
-        Assert.assertEquals(
-                Arrays.asList(new GoAction(SimplifiedRoadMapOfPartOfRomania.BUCHAREST),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.FAGARAS),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.SIBIU),
-                        new GoAction(SimplifiedRoadMapOfPartOfRomania.ARAD)),
-                searchForActions(problemPairtest5.getFirst(), problemPairtest5.getSecond()).fromGoalStateToInitialState());
+    public void timisoaraToUrziceni() throws Exception {
+      //Result will contain all intermediate nodes excluding initial and final nodes.
+        Pair<Problem<GoAction, InState>, Problem<GoAction, InState>> pair =
+            ProblemFactory.getSimpleBidirectionalSearchProblem(TIMISOARA, URZICENI);
+        final BidirectionalActions<GoAction> solution = searchForActions(pair);
+
+        assertEquals(Arrays.asList(
+            new GoAction(ARAD),new GoAction(SIBIU), new GoAction(FAGARAS)),
+            solution.fromInitialStatePart());
+
+        assertEquals(Arrays.asList(new GoAction(BUCHAREST)),
+            solution.fromGoalStatePart());
+
+        assertEquals(Arrays.asList(
+            new GoAction(ARAD),
+            new GoAction(SIBIU),
+            new GoAction(FAGARAS),
+            new GoAction(BUCHAREST)),
+            solution.fromInitialStateToGoalState());
+
+        assertEquals(Arrays.asList(
+            new GoAction(BUCHAREST),
+            new GoAction(FAGARAS),
+            new GoAction(SIBIU),
+            new GoAction(ARAD)),
+            solution.fromGoalStateToInitialState());
     }
 
+    @Test @Ignore("fails, actual is [null, Go(Oradea), null]")
```

Comment:
It is a corner case that needs to be handled. It will make sense if you see ```BasicBidirectionalActions```. As far as the tests are concerned, I will make a PR wherein this corner case will be handled.